packages/fetch/tsup.config.ts

```diff
@@ -0,0 +1,3 @@
+import { createTsupConfig } from '../../scripts/tsup.config';
+
+export default createTsupConfig({ globalName: 'SapphireFetch' });
```

Comment:
Are we sure tsup deals with cross-fetch correctly for UMD?

Reply:
Why do you think I labeled this "needs testing" LMAO.

Reply:
Okay it definitely does not. It goes:
```js
const result = await (0, import_cross_fetch.fetch)(stringUrl, options);
```
`import_cross_fetch.fetch` is
```js
var import_cross_fetch = __toESM(require_node_ponyfill());
```
```js
var require_node_ponyfill = __commonJS({
    "../../node_modules/cross-fetch/dist/node-ponyfill.js"(exports, module) {
      var nodeFetch2 = (init_lib(), __toCommonJS(lib_exports));
      var realFetch = nodeFetch2.default || nodeFetch2;
```

So I'll have to come up with a plan for this.

Reply:
Tested latest commit with
- `index.mjs` (ESM)
- `index.cjs` (CJS)
- iife (through `index.html` with `<script src="./dist/index.global.js></script>` in `head` and a fetch in body)
- webpack (through create-react-app)
- swc (through NextJS)

All finally working. It took reverting back to cross-fetch and rollup which is a big shame but I have no idea what else to do. No matter what I tried, one thing or the other just didn't work. I have no idea how we're ever going to do undici for this package.